### PR TITLE
add support for binary/octet stream in Bind method

### PR DESCRIPTION
**Pull request by Umang01-hash**
## Pull Request Template

**Description:**

-   Closes issue #1143

- #### Example Usage:

```go
func HelloHandler(c *gofr.Context) (interface{}, error) {
	var data []byte
	if err := c.Bind(&data); err != nil {
		return nil, fmt.Errorf("failed to bind binary data: %w", err)
	}

	// Process the binary data
	fmt.Printf("Received binary data: %v\n", data)
	return string(data), nil
}
```

**Checklist:**

-   [x] I have formatted my code using  `goimport`  and  `golangci-lint`.
-   [x] All new code is covered by unit tests.
-   [x] This PR does not decrease the overall code coverage.
-   [x] I have reviewed the code comments and documentation for clarity.

**Thank you for your contribution!**

### CODE REVIEW

File: pkg/gofr/http/request.go [hunk 1]
```diff
@@ -157,3 +160,22 @@ func (r *Request) bindForm(ptr any, isMultipart bool) error {
...
 	return nil
 }
+
+// bindBinary handles binding for binary/octet-stream content type.
+func (r *Request) bindBinary(i interface{}) error {
+	// Ensure i is a pointer to a byte slice
+	byteSlicePtr, ok := i.(*[]byte)
+	if !ok {
+		return fmt.Errorf("%w: %v", errNonSliceBind, i)
+	}
```
Comment (ccoVeille) [inline pkg/gofr/http/request.go:170]: `i` usually state for loop counter

```suggestion
func (r *Request) bindBinary(raw interface{}) error {
	// Ensure i is a pointer to a byte slice
	byteSlicePtr, ok := raw.(*[]byte)
	if !ok {
		return fmt.Errorf("%w: %v", errNonSliceBind, raw)
	}
```

Comment (Umang01-hash) [reply pkg/gofr/http/request.go]: got it! changed the argument name to `raw`

File: pkg/gofr/http/request_test.go [hunk 2]
```diff
@@ -272,3 +273,44 @@ func TestBind_FormURLEncoded(t *testing.T) {
...
+		req: httptest.NewRequest(http.MethodPost, "/binary", http.NoBody),
+	}
+	req.req.Header.Set("Content-Type", "binary/octet-stream")
+
+	err := req.Bind("invalid input")
+	expectedErr := fmt.Errorf("%w: %v", errNonSliceBind, "invalid input")
+
+	if err == nil || !strings.Contains(err.Error(), expectedErr.Error()) {
+		t.Errorf("Expected error: %v, got: %v", expectedErr, err)
+	}
```
[the remark this answers is not in the document]
Comment (Umang01-hash) [reply pkg/gofr/http/request_test.go]: Sure changed the test according to your suggestion

File: pkg/gofr/http/middleware/logger.go [hunk 3]
```diff
@@ -140,6 +140,7 @@ func panicRecovery(re any, w http.ResponseWriter, logger logger) {
 	default:
 		e = "Unknown panic type"
 	}
+
```
Comment (ccoVeille) [inline pkg/gofr/http/middleware/logger.go:143]: Maybe this line could be removed, but I tend do not commit such change, when there is nothing else to change in a file

It helps when looking at file history.

Of course, such change is fine if the MR purpose is of course about formatting

File: pkg/gofr/http/request.go [hunk 4]
```diff
@@ -157,3 +160,22 @@ func (r *Request) bindForm(ptr any, isMultipart bool) error {
 
 	return nil
 }
+
+// bindBinary handles binding for binary/octet-stream content type.
+func (r *Request) bindBinary(raw interface{}) error {
+	// Ensure i is a pointer to a byte slice
```
Comment (ccoVeille) [inline pkg/gofr/http/request.go:166]: You didn't rename `i` here


```suggestion
	// Ensure raw is a pointer to a byte slice
```

**Review by ccoVeille — APPROVED** [review]
Minor remarks, but 👍

File: pkg/gofr/http/middleware/logger.go (hunk 3 above)
Comment (Umang01-hash) [reply pkg/gofr/http/middleware/logger.go:143]: Screenshot 2024-11-20 at 3 22 49 PM

@ccoVeille removing that line will lead to a linter error😐

File: pkg/gofr/http/request.go (hunk 4 above)
Comment (Umang01-hash) [reply pkg/gofr/http/request.go:166]: sure changed

**Review by ccoVeille — APPROVED** [review]

**Review by coolwednesday — APPROVED** [review]

**Review by vikash — APPROVED** [review]

**Pull request merged**